frontend/server/src/Controllers/Contest.php

```diff
@@ -1116,6 +1121,36 @@ public static function getContestEditForTypeScript(
         ];
     }
 
+    /**
+     * Adding the version log to every problem in a contest
+     * @param list<ProblemsetProblem> $problems
+     * @return list<ProblemsetProblemWithVersions> $problems
+     */
+    private static function addVersionsToProblems(
+        array $problems,
+        \OmegaUp\DAO\VO\Identities $identity
+    ): array {
+        $problemsWithVersion = [];
+        foreach ($problems as $problem) {
+            unset($problem['problem_id']);
+            $problemsWithVersion[] = array_merge(
+                [
+                    'versions' => \OmegaUp\Controllers\Problem::getVersions(
+                        new \OmegaUp\DAO\VO\Problems(
+                            array_intersect_key(
+                                $problem,
+                                \OmegaUp\DAO\VO\Problems::FIELD_NAMES
+                            )
+                        ),
+                        $identity
+                    )
+                ],
+                $problem
+            );
```

Comment:
```suggestion
            $problem['versions'] = \OmegaUp\Controllers\Problem::getVersions(
                new \OmegaUp\DAO\VO\Problems(
                    array_intersect_key(
                        $problem,
                        \OmegaUp\DAO\VO\Problems::FIELD_NAMES
                    )
                ),
                $identity
            );
            $problemsWithVersions[] = $problem;
```

esto funciona? es un poco más fácil de leer.

Reply:
Listo